atomicwrites/__init__.py

```diff
@@ -19,12 +19,21 @@ def _path_to_unicode(x):
 
 
 if sys.platform != 'win32':
+    def _sync_directory(dst):
+        fd = os.open(os.path.dirname(dst), 0)
+        try:
+            os.fsync(fd)
+        finally:
+            os.close(fd)
+
     def _replace_atomic(src, dst):
         os.rename(src, dst)
+        _sync_directory(dst)
```

Comment:
You also have to fsync ``src`` if it doesn't share the same parent directory with ``dst``. Although the operation will not be atomic in that case. (The same for ``_move_atomic``)